docs/create-a-business-component.md

```diff
@@ -0,0 +1,345 @@
+
+Before diving into the code, we have to introduce a new concept: **Enhancers**.
+In this case, it will be responsible to fetch the data from our **GraphQL
+schema**, transform the data and feed it to our component.
+
+The **Enhancers** are not means to only be _data fetchers_, they contain most of
+the **Business logic** of our application, we use the
+[Higher-Order Components pattern (HOC)](https://reactjs.org/docs/higher-order-components.html)
+to create them.
+
+> In Front-Commerce, we use a react library:
+> [Recompose](https://github.com/acdlite/recompose) to handle composition of
+> **HOC**, it provides a lots of helpers which are really usefull to _enrich_
+> our components.
+
+In the following steps, we are going to build our store locator. It will be a
+simple marker on a map indicating where to find our store.
+
+### Installing the dependencies
+
+To create the map, we are going to use the `react-leaflet` package. It is a
+simple component which will allow us to display OpenStreetMap easely.
+
+This is one of the biggest advantage of using React to build our front-end, we
+have access to this huge ecosystem.
+
+So we just install the two required packages :
+
+```sh
+npm install react-leaflet leaflet`
+```
+
+### Our new Homepage
+
+You don't need to anticipate every **UI** or **Business** components in your
+application. When your component will begin to get bigger, the need to extract
+some of them will come naturally.
+
+To illustrate this point, we are going to first add a hardcoded version of our
+map into the homepage. Then we will extract it into its own module and finally
+we are going to fetch the required data from the **GraphQl schema**.
+
+Our first working version will look like:
+
+```js
+// src/theme/pages/Home/Home.js;
+
+//... Existing imports
+import { Map, TileLayer, Marker, Popup } from "react-leaflet";
+import "leaflet/dist/leaflet.css";
+import "leaflet/dist/images/marker-icon.png";
+import "leaflet/dist/images/marker-shadow.png";
+
+const Home = ({ store }) => (
+  <div className="page page--home">
+    //... The rest of the homepage
+    <Map
+      center={[43.584296, 1.44182]}
+      zoom={14}
+      style={{ height: "600px", width: "800px" }}
+    >
+      <TileLayer
+        attribution="&amp;copy <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
+        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+      />
+      <Marker position={[43.584296, 1.44182]}>
+        <Popup>
+          <span>My awesome store is HERE!</span>
+        </Popup>
+      </Marker>
+    </Map>
+  </div>
+);
+// ...
+```
+
+With that, you should see the map appears in your homepage.
+
+### Extracting our new component
+
+Having the map in the Home component could be fine for a time, but we don't want
+to maintain big bloated components, so we are going to extract it to its own
+Module.
+
+For it is exactly the same as before, we juste moved it to the
```

Comment:
typo : just~~e~~

Reply:
I think a word (« For **now** » ?) is missing in this sentence.

Reply:
I read it as "Since it is..." or "Because it is..." but maybe the author could clarify :slightly_smiling_face: